main/src/ca/uwaterloo/flix/language/phase/Redundancy.scala

```diff
@@ -636,34 +636,49 @@ object Redundancy extends Phase[TypedAst.Root, TypedAst.Root] {
   }
 
   /**
-    * Returns the symbols used in the given list of pattern `ps`.
-    */
+   * Returns the symbols used in the given list of pattern `ps`.
+   */
   private def visitPats(ps: List[Pattern]): Used = ps.foldLeft(Used.empty) {
     case (acc, pat) => acc ++ visitPat(pat)
   }
 
   /**
-    * Returns the symbols used in the given constraint `c0` under the given environment `env0`.
-    */
+   * Returns the symbols used in the given constraint `c0` under the given environment `env0`.
+   */
   private def visitConstraint(c0: Constraint, env0: Env): Used = {
     val zero = visitHeadPred(c0.head, env0)
     val used = c0.body.foldLeft(zero) {
       case (acc, b) => acc ++ visitBodyPred(b, env0)
     }
-    used -- c0.cparams.map(_.sym)
+
+    val errors = c0.cparams.flatMap(constraintParam => {
+      val sym = constraintParam.sym
+      val occurrences = used.varSyms.apply(sym)
+      if (occurrences.isEmpty) throw InternalCompilerException(s"Variable frequency is zero for $sym")
```

Comment:
I would just omit this check. Yes it should be true, but even if it is not, its not a problem for this phase.